embedx/config.schema.json

```diff
@@ -783,6 +783,80 @@
           ]
         }
       }
+    },
+    "courierTemplates": {
+      "type": "object",
+      "properties": {
+        "invalid": {
+          "type": "object",
+          "properties": {
+            "email": {
+              "$ref": "#/definitions/emailCourierTemplate"
+            }
+          },
+          "required": [
+            "email"
+          ]
+        },
+        "valid": {
+          "type": "object",
+          "properties": {
+            "email": {
+              "$ref": "#/definitions/emailCourierTemplate"
+            }
+          },
+          "required": [
+            "email"
+          ]
+        }
+      }
+    },
+    "emailCourierTemplate": {
+      "type": "object",
+      "properties": {
+        "template_root": {
+          "type": "string",
+          "description": "The entry point for the template when using nested templates. This is optional as the template does not need to define an entry point."
+        },
+        "body": {
+          "type": "object",
+          "properties": {
+            "plaintext": {
+              "type": "string",
+              "description": "The fallback template for email clients that do not support html.",
+              "format": "uri",
+              "examples": [
+                "file://path/to/body.plaintext.gotmpl",
+                "https://foo.bar.com/path/to/body.plaintext.gotmpl",
+                "base64://e3sgZGVmaW5lIGFmLVpBIH19CkhhbGxvLAoKSGVyc3RlbCBqb3UgcmVrZW5pbmcgZGV1ciBoaWVyZGllIHNrYWtlbCB0ZSB2b2xnOgp7ey0gZW5kIC19fQoKe3sgZGVmaW5lIGVuLVVTIH19CkhpLAoKcGxlYXNlIHJlY292ZXIgYWNjZXNzIHRvIHlvdXIgYWNjb3VudCBieSBjbGlja2luZyB0aGUgZm9sbG93aW5nIGxpbms6Cnt7LSBlbmQgLX19Cgp7ey0gaWYgZXEgLmxhbmcgImFmLVpBIiAtfX0KCnt7IHRlbXBsYXRlICJhZi1aQSIgLiB9fQoKe3stIGVsc2UgLX19Cgp7eyB0ZW1wbGF0ZSAiZW4tVVMiIH19Cgp7ey0gZW5kIC19fQp7eyAuUmVjb3ZlcnlVUkwgfX0K"
+              ]
+            },
+            "html": {
+              "type": "string",
+              "description": "The default template used for sending out emails. The template can contain HTML ",
+              "format": "uri",
+              "examples": [
+                "file://path/to/body.html.gotmpl",
+                "https://foo.bar.com/path/to/body.html.gotmpl",
+                "base64://e3sgZGVmaW5lIGFmLVpBIH19CkhhbGxvLAoKSGVyc3RlbCBqb3UgcmVrZW5pbmcgZGV1ciBoaWVyZGllIHNrYWtlbCB0ZSB2b2xnOgp7ey0gZW5kIC19fQoKe3sgZGVmaW5lIGVuLVVTIH19CkhpLAoKcGxlYXNlIHJlY292ZXIgYWNjZXNzIHRvIHlvdXIgYWNjb3VudCBieSBjbGlja2luZyB0aGUgZm9sbG93aW5nIGxpbms6Cnt7LSBlbmQgLX19Cgp7ey0gaWYgZXEgLmxhbmcgImFmLVpBIiAtfX0KCnt7IHRlbXBsYXRlICJhZi1aQSIgLiB9fQoKe3stIGVsc2UgLX19Cgp7eyB0ZW1wbGF0ZSAiZW4tVVMiIH19Cgp7ey0gZW5kIC19fQo8YSBocmVmPSJ7eyAuUmVjb3ZlcnlVUkwgfX0iPnt7IC5SZWNvdmVyeVVSTCB9fTwvYT4"
+              ]
+            }
+          },
+          "required": [
+            "plaintext",
+            "html"
```

Comment:
Are they really required? Wouldn't these fall back to the default if undefined?

Reply:
They would, but they are only required when an `email` key was specified. This forces the user to implement html and plaintext templates, instead of e.g. just html

Reply:
Ok got it, you can probably use `if` statements to check whether one of them is set and then require both :)

Reply:
https://github.com/Benehiko/kratos/blob/ea6a585d9d4a91b53acf971cca74ad7b45bc3e19/embedx/config.schema.json#L817-L844

This is using a definition that does not require the `body`. It only activates the required fields when setting `body`.